daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Classify.scala

```diff
@@ -10,33 +10,33 @@ import com.daml.lf.speedy.SExpr._
 object Classify { // classify the machine state w.r.t what step occurs next
 
   case class Counts(
```

Comment:
I do not like using case class for stateful object. 
It is very annoying for you to use a normal class.

you should also make this object `final`.

Reply:
ok, will do...